### PR TITLE
Add documentation for exit() function

### PR DESCRIPTION
**Pull request by vinayb21**

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

File: Lib/argparse.py [hunk 1]
```diff
@@ -2480,6 +2480,13 @@ def _print_message(self, message, file=None):
     # Exiting methods
     # ===============
     def exit(self, status=0, message=None):
+        """exit(status: integer, message)
+
+        status = 0 is considered as successful termination. Any nonzero exit code is considered as an "abnormal termination"
+
+        if message is None then status is 0 else the status is 1
```
Comment (asottile) [inline Lib/argparse.py]: this part doesn't appear to be true:

```
>>> import argparse
>>> parser = argparse.ArgumentParser()
>>> parser.exit(0, 'hello world')
hello world
$ echo $?
0
```

**Review by eamanu — CHANGES_REQUESTED** [review]
IMO, the docstring improve are good. But I think that this PR has some things to correct:
* pep8
* A short title that briefly summarizes the functionality
* @asottile comments
*  look the other docstring formats.

Please sign CLA. I don't know if is necessary open a issue for this. I want to listen other opinions.

File: Lib/argparse.py [hunk 2]
```diff
@@ -2480,6 +2480,12 @@ def _print_message(self, message, file=None):
     # Exiting methods
     # ===============
     def exit(self, status=0, message=None):
+        """exit(status: integer, message)
+
```
Comment (pablogsal) [inline Lib/argparse.py:2484]: This docstring is missing a short description of what the exit function *does*.

**Review by pablogsal — CHANGES_REQUESTED** [review]
Thank you for the PR! We can only continue the review when the CLA is signed.

In the meanwhile, please, add a NEWS entry using the `blurb` tool.

**Comment by brettcannon** [comment]
Thanks for the PR, but closing as the CLA has not been signed within the last month. If you do decide to sign the CLA we can re-open this PR.

**Pull request closed without merging**